garage/tf/models/cnn_model.py

```diff
@@ -0,0 +1,44 @@
+"""CNN Model."""
+import tensorflow as tf
+
+from garage.tf.core.cnn import cnn
+from garage.tf.models.base import Model
+
+
+class CNNModel(Model):
+    """
+    CNN Model.
+
+    Args:
+        filter_dims: Dimension of the filters.
+        num_filters: Number of filters.
+        strides: The stride of the sliding window.
+        name: Variable scope of the cnn.
+        padding: The type of padding algorithm to use, from "SAME", "VALID".
+        hidden_nonlinearity: Activation function for intermediate dense
+            layer(s).
+    """
+
+    def __init__(self,
+                 filter_dims,
+                 num_filters,
+                 strides,
+                 name=None,
```

Comment:
name='CNNModel'

Reply:
`Model` has a default name if name=None, which will be the class name. How should we do this? If we enforce all derived model class to have name, we don't need the default name at all.

Reply:
Seems like you have already violated the `Model` interface by making `name` an kwarg: https://github.com/rlworkgroup/garage/blob/master/garage/tf/models/base.py#L159

Reply:
oh yes, so a name is actually required.